x/txfees/keeper/hooks_test.go

```diff
@@ -436,24 +434,33 @@ func (s *KeeperTestSuite) TestAfterEpochEnd() {
 	err := s.App.TxFeesKeeper.AfterEpochEnd(s.Ctx, "day", 1)
 	s.Require().NoError(err)
 
-	// Confirm that staking fee collector only has denomWithNoPool and denomWithNoProtorevLink left in balance.
-	validateEndCollectorBalance(stakingCollectorAddress)
+	// Confirm that tx fee collector only has denomWithNoPool and denomWithNoProtorevLink left in balance.
+	txFeeCollectorAddressBalance := s.App.BankKeeper.GetAllBalances(s.Ctx, txFeeCollectorAddress)
+	s.Require().Len(txFeeCollectorAddressBalance, 2)
+	s.Require().Equal(txFeeCollectorAddressBalance[0].Denom, denomWithNoPool)
+	s.Require().Equal(txFeeCollectorAddressBalance[1].Denom, denomWithNoProtorevLink)
 
 	// Confirm that that all native tokens are sent to the fee collector.
-	feeCollectorAddress := s.App.AccountKeeper.GetModuleAddress(types.FeeCollectorName)
+	feeCollectorAddress := s.App.AccountKeeper.GetModuleAddress(authtypes.FeeCollectorName)
 	feeCollectorBalance := s.App.BankKeeper.GetAllBalances(s.Ctx, feeCollectorAddress)
 	s.Require().Len(feeCollectorBalance, 1)
 	s.Require().Equal(feeCollectorBalance[0].Denom, stakingDenom)
 
-	// Confirm that community pool fee collector only has denomWithNoPool and denomWithNoProtorevLink left in balance.
-	validateEndCollectorBalance(communityPoolCollectorAddress)
+	// Confirm that community pool fee collector only denomWithNoProtorevLink left in balance.
+	// denomWithNoPool is a whitelisted asset, so it should be directly sent to the community pool.
+	communityPoolCollectorBalance := s.App.BankKeeper.GetAllBalances(s.Ctx, communityPoolCollectorAddress)
+	s.Require().Len(communityPoolCollectorBalance, 1)
+	s.Require().Equal(communityPoolCollectorBalance[0].Denom, denomWithNoProtorevLink)
 
 	communityPoolBalanceAfter := s.App.BankKeeper.GetAllBalances(s.Ctx, communityPoolAddress)
 	communityPoolBalanceDelta := communityPoolBalanceAfter.Sub(communityPoolBalanceBefore...)
 
 	// Confirm that that all tokens that are of the configured denom parameter are sent to the community pool.
-	s.Require().Len(communityPoolBalanceDelta, 1)
-	s.Require().Equal(communityPoolBalanceDelta[0].Denom, communityPoolDenom)
+	s.Require().Len(communityPoolBalanceDelta, 4)
+	s.Require().Equal(communityPoolBalanceDelta[0].Denom, otherPreSwapDenom)
+	s.Require().Equal(communityPoolBalanceDelta[1].Denom, denomWithNoPool)
+	s.Require().Equal(communityPoolBalanceDelta[2].Denom, preSwapDenom)
+	s.Require().Equal(communityPoolBalanceDelta[3].Denom, communityPoolDenom)
```

Comment:
All these denoms are whitelisted quote assets, so they are direct sends to the community pool. The denomWithNoProtorevLink is not whitelisted, so it stays in the communityPoolCollectorAddress.